python/src/scipp/compat/mantid.py

```diff
@@ -130,21 +150,38 @@ def convert_Workspace2D_to_dataset(ws):
                          coords={
                              dim: coord,
                              sc.Dim.Spectrum: num
-                         },
-                         labels={
-                             "position": pos,
+    },
+        labels={"position": pos,
                              "component_info": comp_info,
                              "detector_info": det_info
-                         },
-                         attrs={
-                             "run": make_run(ws),
+                },
+        attrs={"run": make_run(ws),
                              "sample": make_sample(ws)
-                         })
+               },
+        masks={"bin": make_bin_masks(common_bins,
+                                     dim,
+                                     ws.blocksize(),
+                                     ws.getNumberHistograms()),
+               "spectrum": sc.Variable([sc.Dim.Spectrum],
+                                       shape=(ws.getNumberHistograms(),),
+                                       dtype=sc.dtype.bool)
+               })
```

Comment:
Why are you always settings masks? Usually there should be no masks, or maybe only one of them.